client/src/main/java/com/tigrisdata/db/client/config/TigrisConfiguration.java

```diff
@@ -96,11 +106,91 @@ public Builder withObjectMapper(ObjectMapper objectMapper) {
       return this;
     }
 
+    /**
+     * This will enable and customize {@link OAuth2Config}.
+     *
+     * @param oAuth2Config oauth2 config
+     * @return ongoing builder
+     */
+    public Builder withOAuth2(OAuth2Config oAuth2Config) {
+      this.oAuth2Config = oAuth2Config;
+      return this;
+    }
+
     public TigrisConfiguration build() {
       return new TigrisConfiguration(this);
     }
   }
 
+  public static class OAuth2Config {
+    private final String refreshToken;
+    private final String clientId;
```

Comment:
ClientID and ClientPassword are unused in the case of refresh token, I guess.

Reply:
clientsecret is not required but client id is required for refreshing access token.

Reply:
I think access and refresh token is all what is needed:
https://datatracker.ietf.org/doc/html/rfc6749#page-47

Reply:
we discussed this further and decided to proxy external calls via server.